scripts/cardboardlinter

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python
+
+import argparse
+import subprocess
+import sys
+
+import yaml
+from cardboardlint.linter_cppcheck import linter_cppcheck
+from cardboardlint.linter_import import linter_import
+from cardboardlint.linter_namespace import linter_namespace
+from cardboardlint.linter_pylint import linter_pylint
+
+
+def main():
+    # get arguments
+    args = parse_args()
+
+    # get git diff
+    if args.refspec is None:
+        files_lines = None
+    else:
+        files_lines = run_diff(args.refspec)
+
+    # load lint configuration for module
+    config = load_config('.cardboardlint.yml')
+
+    # select specific linter if desired
+    if args.selections is None:
+        config_linters = config
+    else:
+        config_linters = {selection: config[selection] for selection in args.selections}
+
+    returncode = 0
+    for lintname, lintconfig in config_linters.items():
+        if lintname not in LINTERS:
+            continue
+        messages = [message for message in LINTERS[lintname](lintconfig, files_lines)
+                    if message.indiff(files_lines)]
+        if len(messages) > 0:
+            returncode = -1
+        for message in messages:
+            print(message)
+    sys.exit(returncode)
+
+
+def parse_args():
+    """Parse the arguments given to the script."""
+    parser = argparse.ArgumentParser(prog='cardboardlint')
+    parser.add_argument('--refspec', dest='refspec', nargs='?', default=None,
+                        help=('The parent commit used to compute the diff. If given, only the files'
+                              ' that have been changed will be compared. Otherwise, all files are '
+                              'compared.'))
+    # The next argument should become a lot smarter in future, e.g. select all static or
+    # dynamic linters, or a fraction of them, e.g. "dynamic 0%-33%".
+    parser.add_argument('--selections', dest='selections', nargs='*', default=None,
+                        help='Run just the given linters.')
+    return parser.parse_args()
+
+
+def run_diff(refspec_parent):
+    """Run git diff with respect to current branch
+
+    Parameters
+    ----------
+    refspec_parent : str
+        Reference to the parent branch
+
+    Returns
+    -------
+    files_lines : dict
+        Dictionary of filename to the set of line numbers (that have been modified)
+    """
+    # generate 0 lines of context (i.e. only the lines that differ)
+    # FIXME: gitpython?
```

Comment:
I'd rather avoid the dependency on gitpython. This is the only place where we would need it, but the benefit is too shallow to add a dependency IMHO.